File: src/types/publish.ts
import type { SecurityIssue } from './security';

export interface CheckStatus {
  type: 'Pass' | 'Fail' | 'Warning';
}

export type CheckStatusType = 'Pass' | 'Fail' | 'Warning';

export interface PreflightCheck {
  name: string;
  status: CheckStatusType;
  message: string;
}

export interface SecurityReport {
  score: number;
  level: 'Safe' | 'Low' | 'Medium' | 'High' | 'Critical';
  issues: SecurityIssue[];
  blocked: boolean;
}

export interface PreflightResult {
  passed: boolean;
  checks: PreflightCheck[];
  security_report?: SecurityReport;
}

export interface PublishResult {
  success: boolean;
  message: string;
  skill_id?: string;
  listing_id?: string;
  published_at?: number;
}

export interface PublishMetadata {
  name: string;
  description: string;
  author: string;
  version: string;
  tags: string[];
}

export type PublishStatus = 'Published' | 'Failed' | 'Pending';

export interface PublishRecord {
  id: string;
  skill_name: string;
  skill_version: string;
  skill_id: string;
  listing_id: string;
  author: string | null;
  description: string | null;
  tags: string[];
  published_at: number;
  status: PublishStatus;
  error_message: string | null;
}

File: src-tauri/src/models/publish.rs
use serde::{Deserialize, Serialize};

/// 发布历史记录
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct PublishRecord {
    pub id: String,
    pub skill_name: String,
    pub skill_version: String,
    pub skill_id: String,
    pub listing_id: String,
    pub author: Option<String>,
    pub description: Option<String>,
    pub tags: Vec<String>,
    pub published_at: i64,      // Unix timestamp (milliseconds)
    pub status: PublishStatus,   // published, failed, pending
    pub error_message: Option<String>,
}

/// 发布状态
#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
pub enum PublishStatus {
    Published,
    Failed,
    Pending,
}
